base/services/w32time/w32time.c

```diff
@@ -282,26 +284,40 @@ W32TmServiceMain(DWORD argc, LPWSTR *argv)
     /* The worker loop of a service */
     for (;;)
     {
-        result = SetTime();
+        /* Is it Auto-Sync? */
+        bAutoSync = FALSE;
+        if (RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                          L"SYSTEM\\CurrentControlSet\\Services\\W32Time\\Parameters",
+                          0,
+                          KEY_QUERY_VALUE,
+                          &hKey) == ERROR_SUCCESS)
+        {
+            cbData = sizeof(szData);
+            RegQueryValueExW(hKey, L"Type", NULL, NULL, (LPBYTE)szData, &cbData);
+            szData[ARRAYSIZE(szData) - 1] = UNICODE_NULL; /* Avoid buffer overrun */
+            bAutoSync = (wcscmp(szData, L"NTP") == 0);
```

Comment:
what happens on windows if "NTP" is specified in e.g. lowercase? Would it also do the autosync?

Reply:
Why favoring ARRAYSIZE over _countof? I perceived both to be equivalent in their meaning (and typically favor _countof because of it being a character less).

Reply:
_countof requires the extra inclusion of stdlib.h or a similar CRT header. So if that one hasn't been included already (because code didn't use other functionality from it), it may be seen as overkill.

Reply:
Right. It would be massive overkill to add an include to save a keystroke. Thx, **my** question has been answered. Yours not yet.